### PR TITLE
[KYUUBI #6476] Fix incomplete app events deserialization in SHS

### LINKED ISSUE #6476: [Bug] spark historyserver Show incomplete applications kyuubi query engine ui error
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

spark historyserver Show incomplete applications kyuubi query engine ui error

```
<html>
<body>
<!--StartFragment--><h2 style="color: rgb(0, 0, 0); font-family: &quot;Microsoft YaHei&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">HTTP ERROR 500 java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.Long</h2>
URI: | /history/application_1698577744226_0074/kyuubi/
-- | --
500
java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.Long
org.apache.spark.ui.JettyUtils$$anon$1-1fc4c49c
java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.Long

<h3 style="color: rgb(0, 0, 0); font-family: &quot;Microsoft YaHei&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Caused by:</h3><pre style="color: rgb(0, 0, 0); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.Long
	at scala.runtime.BoxesRunTime.unboxToLong(BoxesRunTime.java:107)
	at org.apache.spark.ui.StatementStatsPagedTable.row(EnginePage.scala:542)
	at org.apache.spark.ui.StatementStatsPagedTable.row(EnginePage.scala:452)
	at org.apache.spark.ui.PagedTable.$anonfun$table$1(PagedTable.scala:127)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
	at scala.collection.TraversableLike.map(TraversableLike.scala:286)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:279)</pre><!--EndFragment-->
</body>
</html>
```

### Affects Version(s)

master/1.9.1

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by felixzh2020**
# :mag: Description
## Issue References 🔗

This pull request fixes #6476 : spark historyserver -> Show incomplete applications -> kyuubi query engine ui error（java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.Long）.

The reason:
it's related to https://github.com/FasterXML/jackson-module-scala/issues/62

## Describe Your Solution 🔧

add @JsonDeserialize(contentAs = classOf[java.lang.Long]) annotation

## Types of changes :bookmark:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/events/SparkOperationEvent.scala [hunk 1]
```diff
@@ -67,7 +67,9 @@ case class SparkOperationEvent(
     sessionId: String,
     sessionUser: String,
     executionId: Option[Long],
```
Comment (wForget) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/events/SparkOperationEvent.scala:69]: Also add an annotation for `executionId`.

Comment (felixzh2020) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/events/SparkOperationEvent.scala:70]: ok

**Review by wForget — APPROVED** [review]
LGTM, thanks for this fix. Please fix the pr title and complete checklist.

**Comment by felixzh2020** [comment]
"fix the pr title"  is finished.
Failed checks seems unrelated to the content I modified.  Is there a tool similar to flink project "@flinkbot run azure" available？
May be rerun check.

**Comment by wForget** [comment]
> "fix the pr title" is finished. Failed checks seems unrelated to the content I modified. Is there a tool similar to flink project "@flinkbot run azure" available？ May be rerun check.

Never mind, that's an unrelated failure.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master/1.9.2

**Pull request closed without merging**